### ISSUE TITLE
REBRANDING | ARMCORD -> LEGCORD

### ISSUE THREAD

**Issue by SomeAspy** [smartfrigde/armcord#745]
We have received a cease and desist from ARM LLC.
They are requiring us to remove `Arm` from ArmCord.
We are currently working on making changes to package distribution platforms to reflect this.

See the notice at https://public.aspy.dev/000-DMCA-INFO/RECORDS/2024.10.10-Cease_and_Desist-ArmCord-redacted-1-1-1.pdf

Questions or comments? Join the Discord server!
https://discord.gg/pA7DeyHHxu

- [x] armcord.app -> legcord.app
- [x] Notify AUR packagers
	- [x] [armcord-git](https://aur.archlinux.org/packages/armcord-git) -> [legcord-git](https://aur.archlinux.org/packages/legcord-git)
	- [x] [armcord-bin](https://aur.archlinux.org/packages/armcord-bin) -> [legcord](https://aur.archlinux.org/packages/legcord)
	- [x] [armcord](https://aur.archlinux.org/packages/armcord) -> [legcord](https://aur.archlinux.org/packages/legcord)
	- [x] [legcord](https://aur.archlinux.org/packages/legcord) ~~(points to old legcord)~~ Ownership Aquired
- [ ] Update Snap package - https://snapcraft.io/legcord (currently non-functional, waiting for https://github.com/electron-userland/electron-builder/pull/8549)
- [ ] Update Flatpak package
- [x] Update winget package
- [x] Update Scoop package
- [ ] Host new APT repository
- [x] Nix Packages

**Comment by Metrokoto** [comment]
This notice is questionable and potentially illegal.
I didn't want to word anything wrong, so I asked ChatGPT to summarize my thoughts on the matter:

"Common words, phrases, and descriptive terms cannot be trademarked or copyrighted because they belong to the public domain. Trademark law is designed to protect distinctive logos, names, or symbols that identify the source of goods or services, while copyright protects creative works like books, music, or films. However, when a word is too generic or commonly used, it cannot be trademarked because it lacks the uniqueness required to associate it with a specific business or product. Similarly, copyright law does not protect individual words or short phrases, as they are considered basic building blocks of language.

Given this well-established legal principle, any cease and desist notice issued by ARM LLC—if based solely on the use of common terms or descriptive language—would likely be invalid. When a company like ARM issues such notices, especially against open-source developers, it can be seen as an abuse of its legal position. The aim seems less about protecting legitimate intellectual property and more about intimidating smaller entities that lack the resources to defend themselves, thus amounting to borderline bullying or harassment.

Issuing these notices without a valid legal basis not only undermines the principles of fair competition and free speech, but it also tarnishes the credibility of companies that resort to such tactics. It is crucial to respect the boundaries of IP law, particularly in the open-source community, where collaboration and innovation thrive on the open use of language and ideas."

**Disclaimer:** 
**I am not a lawyer but I do know a fair bit about copyright and trademarks and the ChatGPT written statement above is solely to express my personal views on the topic and do not constitute legal advice.**

**Comment by SomeAspy** [comment]
We do not have the funds nor expertise to bring this to court

**Comment by Metrokoto** [comment]
> We do not have the funds nor expertise to bring this to court

You see, this is the problem nowadays, situations like this are rigged from the beginning to favour those with more wealth. 

The same thing here is what happened to Ryuginx, Yuzu, and Palworld from Nintendo... and it really urks me that these tech giants can just bully us into submission like this and get away with it.

@smartfrigde @SomeAspy I would recommend making contact with a YouTuber, if possible, so that the world can see that ARM is just as bad as Nintendo and can be held accountable by the wider public.

I'm sure Louis, Charlie or Mutahar would love to cover this blatant abuse of the system :)

**Comment by SomeAspy** [comment]
This is not the first time.
https://www.theregister.com/2023/08/31/a_star_star_domains/
https://www.theregister.com/2018/07/10/arm_riscv_website/

I am still working with ARM's legal team to find potential solutions to this, but I am saddened that this may be how it is.

**Comment by aidenfoxivey** [comment]
In the process of changing out the icon, I think it looks a bit different on MacOS now.

![CleanShot 2024-10-13 at 21 46 37@2x](https://github.com/user-attachments/assets/f58e0876-ff7a-45ff-9083-7323835a4318)

Would it be acceptable to find a way to make it fit the same borders that other icons use? Happy to help with this if it's desired.

**Comment by aidenfoxivey** [comment]
btw, thank you so much for making legcord. I really appreciate all the hard work on it. I use it daily. :)

**Comment by aidenfoxivey** [comment]
![CleanShot 2024-10-13 at 22 17 58@2x](https://github.com/user-attachments/assets/9bc417b8-e51b-4da0-91ce-4616fe228693)

**Comment by aidenfoxivey** [comment]
Seems you can just take the original file and then use https://github.com/alptugan/icns-creator to make something that follows the design principles.

**Comment by smartfrigde** [comment]
> ![CleanShot 2024-10-13 at 22 17 58@2x](https://private-user-images.githubusercontent.com/71291879/376091327-9bc417b8-e51b-4da0-91ce-4616fe228693.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MzczNTcsIm5iZiI6MTcyODkzNzA1NywicGF0aCI6Ii83MTI5MTg3OS8zNzYwOTEzMjctOWJjNDE3YjgtZTUxYi00ZGEwLTkxY2UtNDYxNmZlMjI4NjkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDIwMTczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwYTI4NzI1YmFkMWI3MDIyZmQ2NDM3YjRiYjY3ZTU1NmEyY2Q3ZjE5ZjgxYWMxYTU4NDFkMzFjOGRkZTcyNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.myCBiPYoYmPggYhmstdNbE735XlMLtaQWVhpRUY8Oz0)

@aidenfoxivey could you provide the icns file you used in this screenshot? I would be happy to add it upstream

**Comment by aidenfoxivey** [comment]
@smartfrigde absolutely! that said, I'm not entirely sure GitHub will let me just link it right here. is there a way you'd prefer it sent over to you?

**Comment by smartfrigde** [comment]
@aidenfoxivey you can dm me on discord, my tag is `smartfrigde` or use any filehost

**Comment by Metrokoto** [comment]
Uhm, just wondering, how was this missed...?
![image](https://github.com/user-attachments/assets/d25b60dd-c854-4a7d-937c-3b899a5de032)

**Comment by aidenfoxivey** [comment]
Is it possibly from here? https://github.com/Legcord/Legcord/blob/2c391a8b288f81d2565610875642782ea132e14b/src/discord/preload/preload.mts#L86

**Comment by smartfrigde** [comment]
> Uhm, just wondering, how was this missed...? ![image](https://private-user-images.githubusercontent.com/161226960/376412646-d25b60dd-c854-4a7d-937c-3b899a5de032.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwMTcxMjcsIm5iZiI6MTcyOTAxNjgyNywicGF0aCI6Ii8xNjEyMjY5NjAvMzc2NDEyNjQ2LWQyNWI2MGRkLWM4NTQtNGE3ZC05MzdjLTNiODk5YTVkZTAzMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNVQxODI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTViMjcxMjVmNjI1NjQ1Y2U4YjI3MTQ0OGQ2MmZiY2MwNTg4MDYyNGIyMzhjMDMyNzY2OGY0MWI5NjdjZWY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nv4_TqPNdbJXJHHneaEW4hXEQ2j4Jwo3KpP-SfN2OjE)

@Metrokoto you probably have `clientName` value set in your config from previous versions
https://github.com/Legcord/Legcord/blob/dev/src/discord/tray.ts#L30

**Comment by NGSpace** [comment]
What a bullshit claim, the "ARM" name is wayyy too popular to be copyrightable.

I really hope some YouTuber or journalist notices this and gives them a taste of bad PR.

**Comment by thenameisluk** [comment]
>     * [ ]  Host new APT repository
would it be possible to get legcord into main/contrib debian repo?

**Comment by JacobKfromIRC** [comment]
Somewhat off-topic at this point I guess, but:

@Metrokoto Regarding the legality of the trademark, I think "ARM" can be a valid trademark because the common word "arm" is unrelated to computers, kind of like how "Apple" can be a valid trademark even though "apple" is also a common word.

I found this read interesting: https://www.justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks/

I also could be wrong though.

**Comment by Metrokoto** [comment]
> Somewhat off-topic at this point I guess, but:
> 
> @Metrokoto Regarding the legality of the trademark, I think "ARM" can be a valid trademark because the common word "arm" is unrelated to computers, kind of like how "Apple" can be a valid trademark even though "apple" is also a common word.
> 
> I found this read interesting: https://www.justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks/
> 
> I also could be wrong though.

@JacobKfromIRC When it comes to those kinds of trademarks, only the logo can be trademarked, as it has been deemed by most courts that any word in the English dictionary is exempt from being trademarked. 

For example, if you launch a microchip manufacturing company called "Micro Solutions Inc.", Microsoft would not be able to sue you successfully just because you have "Micro" in your brand name as it is protected by the "in the dictionary" logic.

TLDR: You cannot copyright or trademark a word in the dictionary.

Regardless, ARM knows full well they won't win in court 
but @SomeAspy and @smartfrigde don't have the money to sue them for false C&D, and that is why they resorted to bullying.

See: https://bestlifeonline.com/trademark-failures

**Comment by thenameisluk** [comment]
> Regardless, ARM knows full well they won't win in court but @SomeAspy and @smartfrigde don't have the money to sue them for false C&D, and that is why they resorted to bullying.
> 
> See: https://bestlifeonline.com/trademark-failures

why do they care anyway about some opensource project anyway?

**Comment by SomeAspy** [comment]
> > Somewhat off-topic at this point I guess, but:
> > @Metrokoto Regarding the legality of the trademark, I think "ARM" can be a valid trademark because the common word "arm" is unrelated to computers, kind of like how "Apple" can be a valid trademark even though "apple" is also a common word.
> > I found this read interesting: [justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks](https://www.justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks/)
> > I also could be wrong though.
> 
> @JacobKfromIRC When it comes to those kinds of trademarks, only the logo can be trademarked, as it has been deemed by most courts that any word in the English dictionary is exempt from being trademarked.
> 
> For example, if you launch a microchip manufacturing company called "Micro Solutions Inc.", Microsoft would not be able to sue you successfully just because you have "Micro" in your brand name as it is protected by the "in the dictionary" logic.
> 
> TLDR: You cannot copyright or trademark a word in the dictionary.
> 
> Regardless, ARM knows full well they won't win in court but @SomeAspy and @smartfrigde don't have the money to sue them for false C&D, and that is why they resorted to bullying.
> 
> See: [bestlifeonline.com/trademark-failures](https://bestlifeonline.com/trademark-failures)

We have no grounds.
![image](https://github.com/user-attachments/assets/23bddb74-fd61-4870-84bb-3a74ddb13108)

**Comment by Screampuff** [comment]
> > > Somewhat off-topic at this point I guess, but:
> > > @Metrokoto Regarding the legality of the trademark, I think "ARM" can be a valid trademark because the common word "arm" is unrelated to computers, kind of like how "Apple" can be a valid trademark even though "apple" is also a common word.
> > > I found this read interesting: [justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks](https://www.justia.com/intellectual-property/trademarks/strength-of-marks/arbitrary-marks/)
> > > I also could be wrong though.
> > 
> > 
> > @JacobKfromIRC When it comes to those kinds of trademarks, only the logo can be trademarked, as it has been deemed by most courts that any word in the English dictionary is exempt from being trademarked.
> > For example, if you launch a microchip manufacturing company called "Micro Solutions Inc.", Microsoft would not be able to sue you successfully just because you have "Micro" in your brand name as it is protected by the "in the dictionary" logic.
> > TLDR: You cannot copyright or trademark a word in the dictionary.
> > Regardless, ARM knows full well they won't win in court but @SomeAspy and @smartfrigde don't have the money to sue them for false C&D, and that is why they resorted to bullying.
> > See: [bestlifeonline.com/trademark-failures](https://bestlifeonline.com/trademark-failures)
> 
> We have no grounds. ![image](https://private-user-images.githubusercontent.com/33640860/378602732-23bddb74-fd61-4870-84bb-3a74ddb13108.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2OTkwMzIsIm5iZiI6MTcyOTY5ODczMiwicGF0aCI6Ii8zMzY0MDg2MC8zNzg2MDI3MzItMjNiZGRiNzQtZmQ2MS00ODcwLTg0YmItM2E3NGRkYjEzMTA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDE1NTIxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzMzc3Mjk5YjViZmNjY2QwYjQ3OTUxZjk1OWQ2YTc5ZjUwNjkwNzZhOGZhM2VjNmI4MDA2YTNkNDRjMjc1ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qVJLtK7Mo6hK05wwWre_-A5TygDKU9mZRFO_YMxJWSY)

No leg to stand on?

(Sorry)

**Comment by otterscript** [comment]
Legcord is a hilarious name, but I had an idea. What about RISCcord? ARM is just a RISC architecture, and maybe one day the program can be ported to run on RISC-V if that becomes popular.

**Comment by SomeAspy** [comment]
Thats in the plans - I have a vague roadmap for it, but I haven't run with it yet

**Comment by SomeAspy** [comment]
> > ```
> > * [ ]  Host new APT repository
> > ```
> 
> would it be possible to get legcord into main/contrib debian repo?

@thenameisluk I do not think so. I couldn't find any details about this and it seemes you need to be trusted by debian.

**Comment by thenameisluk** [comment]
> > > ```
> 
> > > * [ ]  Host new APT repository
> 
> > > ```
> 
> > 
> 
> > would it be possible to get legcord into main/contrib debian repo?
> 
> 
> 
> @thenameisluk I do not think so. I couldn't find any details about this and it seemes you need to be trusted by debian.

@SomeAspy 
there is guide on that [here](https://www.debian.org/doc/manuals/developers-reference/new-maintainer.html)

to get a package into debian you just need to get a sponsor that will upload package to debian archive

shouldn't be that hard

**Comment by nothingneko** [comment]
Not sure if this got brought up, we've renamed it downstream in Terra

See: [terrapkg/packages #2256](https://github.com/terrapkg/packages/pull/2256)

Backport to 40: [terrapkg/packages #2264](https://github.com/terrapkg/packages/pull/2264)
Backport to 41 [terrapkg/packages #2265](https://github.com/terrapkg/packages/pull/2265)

**Comment by sitiom** [comment]
I have now updated the Scoop package accordingly:
- https://github.com/ScoopInstaller/Extras/pull/14323

**Comment by qwertychouskie** [comment]
> Legcord is a hilarious name, but I had an idea. What about RISCcord? ARM is just a RISC architecture, and maybe one day the program can be ported to run on RISC-V if that becomes popular.

RISCord, I like it.  A clever nod to the original purpose, but also just so happens to line up excellently with "Discord".

**Comment by SomeAspy** [comment]
see but like, we don't have RISC support yet, thats something I want to do but it would require a lot of effort

**Comment by qwertychouskie** [comment]
> see but like, we don't have RISC support yet, thats something I want to do but it would require a lot of effort

Arm is RISC.  RISC is a category that includes Arm, RISC-V, and others.

**Comment by thenameisluk** [comment]
> 
> @SomeAspy 
> there is guide on that [here](https://www.debian.org/doc/manuals/developers-reference/new-maintainer.html)
> 

not sure if you tried to contact them but if it can't get into repo

apt repo can be hosted on github 've already done that https://github.com/thenameisluk/velvet-repo
can just base it on my work

**Comment by SomeAspy** [comment]
> > see but like, we don't have RISC support yet, thats something I want to do but it would require a lot of effort
> 
> Arm is RISC. RISC is a category that includes Arm, RISC-V, and others.

eh. too late now. would be a pain to rebrand again. (package maintainers will lose their shit)

**Comment by knd775** [comment]
Heads up, download links on the website are broken. They all contain `VERSIONNUM` instead of the actual version number. (I apologize if you already knew this!)

**Comment by Truinto** [comment]
Ah, that's why. Trademark is weird. Companies are actually required to sue, so they can keep their trademark. So even if it is dumb, it's more the laws fault. And you can actually trademark works in the dictionary. Just needs to be out of category. See Apple.

I like the rebrand. Good job.

**Comment by NGSpace** [comment]
What trademark? the word "Arm" can't be trademarked...

**Comment by knd775** [comment]
> What trademark? the word "Arm" can't be trademarked...

Take it up with the US patent office then, because it's definitely trademarked.

**Comment by NGSpace** [comment]
> > What trademark? the word "Arm" can't be trademarked...
> 
> Take it up with the US patent office then, because it's definitely trademarked.

I'm not a lawyer (or even american) but I am pretty sure that's the job of a judge.

**Comment by SomeAspy** [comment]
Please limit further discussion of unrelated details (ex. patent law, name suggestions, etc.) to the discord server: https://discord.gg/pA7DeyHHxu

There will be no further rebranding now that we have rebranded to Legcord - I appreciate all the ideas, but rebranding takes a lot of effort for maintainers and developers.

**Comment by retropele** [comment]
Please feel free to resolve comment if this is irrelevant and/or I can open a new thread.

Obviously this change has broken my updates on Ubuntu, which can't be helped.

However, the armcord github hasn't been updated to notify people of the change to Legcord, or to remove install/updater instructions, so I just added the old repo to my sources not knowing it was deprecated.

Does the cease and desist prevent the armcord github from linking/mentioning legcord? There's no mention of the rebranding or relocation on the old github at all. The only indication is the fork notif at the top.

```
retropele@hurtmachina:~$ sudo apt update
Hit:1 http://security.ubuntu.com/ubuntu noble-security InRelease
Hit:2 http://archive.ubuntu.com/ubuntu noble InRelease                   
Hit:3 http://archive.ubuntu.com/ubuntu noble-updates InRelease           
Ign:4 https://apt.armcord.app stable InRelease                           
Ign:4 https://apt.armcord.app stable InRelease    
Ign:4 https://apt.armcord.app stable InRelease
Err:4 https://apt.armcord.app stable InRelease
  Could not handshake: A TLS fatal alert has been received. [IP: [REDACTED] 443]
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up-to-date.
W: Failed to fetch https://apt.armcord.app/dists/stable/InRelease  Could not handshake: A TLS fatal alert has been received. [IP: [REDACTED] 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by NGSpace** [comment]
Any update on the flatpak package? I'm still stuck on an old version of discord.

**Comment by SomeAspy** [comment]
I do not recall what was blocking flatpaks, I think it was something with the github actions

**Comment by NGSpace** [comment]
> I do not recall what was blocking flatpaks, I think it was something with the github actions

Any estimate for a fix? Things are kinda starting to break, insignificant things but still.

**Comment by arrowgent** [comment]
https://github.com/Legcord/Legcord/wiki

wiki needs updates. `Arm -> Leg`

im not sure how to edit or fork to get access to `/wiki` to do a PR so ill just put the note here...

**Comment by smartfrigde** [comment]
If you have any experience with Flatpaks, please see https://github.com/Legcord/Legcord/issues/866

**Comment by arefdsg** [comment]
> > We do not have the funds nor expertise to bring this to court
> 
> You see, this is the problem nowadays, situations like this are rigged from the beginning to favour those with more wealth.
> 
> The same thing here is what happened to Ryuginx, Yuzu, and Palworld from Nintendo... and it really urks me that these tech giants can just bully us into submission like this and get away with it.
> 
> [@smartfrigde](https://github.com/smartfrigde) [@SomeAspy](https://github.com/SomeAspy) I would recommend making contact with a YouTuber, if possible, so that the world can see that ARM is just as bad as Nintendo and can be held accountable by the wider public.
> 
> I'm sure Louis, Charlie or Mutahar would love to cover this blatant abuse of the system :)

Yuzu was an entirely different situation, they deserve what happened to them

**Comment by imide** [comment]
> Yuzu was an entirely different situation, they deserve what happened to them

1. no need to stir up drama from last year
2. u are flat our wrong

**Comment by SomeAspy** [comment]
right thats enough now
